modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/matrix/stats/InternalMatrixStats.java

```diff
@@ -157,7 +157,7 @@ MatrixStatsResults getResults() {
 
     @Override
     public XContentBuilder doXContentBody(XContentBuilder builder, Params params) throws IOException {
-        if (results != null && results.getFieldCounts().keySet().isEmpty() == false) {
+        if (results != null) {
```

Comment:
Looking at `InternalMatrixStats#doReduce()` it seems to me that at least after the aggregations are reduced (which I think should be always the case for aggregations that are rendered to REST) the `results` should always be set. Maybe we can check with somebody who knows more about the internals and work with that assumption on the client side instead of making this change.